examples/metrics/src/http_service.rs

```diff
@@ -18,45 +18,66 @@
 // FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
 // DEALINGS IN THE SOFTWARE.
 
-use hyper::http::StatusCode;
-use hyper::service::Service;
-use hyper::{Body, Method, Request, Response, Server};
+use axum::extract::State;
+use axum::http::StatusCode;
+use axum::response::Response;
+use axum::routing::get;
+use axum::Router;
 use prometheus_client::encoding::text::encode;
 use prometheus_client::registry::Registry;
-use std::future::Future;
-use std::pin::Pin;
+use std::net::SocketAddr;
 use std::sync::{Arc, Mutex};
-use std::task::{Context, Poll};
+use tokio::net::TcpListener;
 
 const METRICS_CONTENT_TYPE: &str = "application/openmetrics-text;charset=utf-8;version=1.0.0";
 
 pub(crate) async fn metrics_server(registry: Registry) -> Result<(), std::io::Error> {
     // Serve on localhost.
-    let addr = ([127, 0, 0, 1], 0).into();
-
-    let server = Server::bind(&addr).serve(MakeMetricService::new(registry));
-    tracing::info!(metrics_server=%format!("http://{}/metrics", server.local_addr()));
-    if let Err(e) = server.await {
-        tracing::error!("server error: {}", e);
-    }
+    let addr: SocketAddr = ([127, 0, 0, 1], 0).into();
+    let service = MetricService::new(registry);
+    let server = Router::new()
+        .route("/metrics", get(respond_with_metrics))
+        .fallback(respond_with_404_not_found)
```

Comment:
I don't think we need an explicit fallback. Does `axum` not do that for us?

Reply:
Haven't worked with `axum` for a while now, I'm not sure about whether it provides default fallback. I simply direct that to existing implementation.

Reply:
A quick search on `docs.rs` returned no info on default fallback. Not sure if `axum` provides such thing. Also we have a custom message for 404 response.

Reply:
I ran the example `axum` setup and found that `axum` made minimal 404 NOT_FOUND response to routes that did not match anything. Is that acceptable?

Reply:
Yeah that is totally fine. These are examples after all so I think less code is better. Users can make their own fallback if they decide to use this.